src/device-registry/validators/readings.validators.js

```
@@ -0,0 +1,422 @@
+            .map((v) => (isValidObjectId(v) ? ObjectId(v) : null))
+            .filter((v) => v !== null);
+        }
+        return value;
+      });
+  },
+  pagination: (defaultLimit = 1000, maxLimit = 2000) => {
+    return (req, res, next) => {
+      let limit = parseInt(req.query.limit, 10);
+      const skip = parseInt(req.query.skip, 10);
+
+      if (isNaN(limit) || limit < 1) {
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)